lib/matplotlib/figure.py

```diff
@@ -3430,8 +3430,11 @@ def tight_layout(self, *, pad=1.08, h_pad=None, w_pad=None, rect=None):
         engine = TightLayoutEngine(pad=pad, h_pad=h_pad, w_pad=w_pad,
                                    rect=rect)
         try:
+            previous_engine = self.get_layout_engine()
             self.set_layout_engine(engine)
             engine.execute(self)
+            if isinstance(previous_engine, ConstrainedLayoutEngine):
```

Comment:
```suggestion
            if not isinstance(previous_engine, TightLayoutEngine):
```
Might be a bit safer to check that the previous engine wasn't Tight, just in case we add another layout engine in the future.